packages/stores/src/derived-data/pool/bonding.ts

```diff
@@ -133,161 +130,196 @@ export class ObservablePoolBonding {
       .sort((a, b) => b - a)
       .reverse()
       .map((durationMs) => {
-        const curDuration = dayjs.duration({
-          milliseconds: durationMs,
-        });
-        const lockedUserShares = queryLockedCoin.getLockedCoinWithDuration(
-          this.poolDetail.poolShareCurrency,
-          curDuration
-        ).amount;
-
-        const userLockedShareValue = this.poolDetail.totalValueLocked.mul(
-          new IntPretty(lockedUserShares.quo(_queryPool.totalShare))
-        );
+        return this.getBondDuration(durationMs);
+      })
+      .filter((duration) => duration !== undefined) as BondDuration[];
+  }
+
+  /** Highest APR that can be earned in this share pool. */
+  get highestBondDuration(): BondDuration | undefined {
+    if (!this.poolDetail.longestDuration) return;
+
+    return this.getBondDuration(
+      this.poolDetail.longestDuration.asMilliseconds()
+    );
+  }
 
-        /** There is only one internal gauge of a chain-configured lockable duration (1,7,14 days). */
-        const internalGaugeOfDuration = internalGauges.find(
-          (gauge) => gauge.duration.asMilliseconds() === durationMs
+  /** Memoizes calculation of bond duration data per duration in ms */
+  protected readonly getBondDuration = computedFn(
```

Comment:
This seems like a fairly complex function to me. Would it be valuable to unit test this?

Reply:
Yeah I think it could, but I simply don't have time to add those right now.